### ISSUE TITLE
[Arm64/Unix] Crossgen Assertion failed 'ldftnToken == targetMethod->gtFptrVal.gtLdftnResolvedToken'

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#10774]
When running crossgen on all *.dll in overlay, ~50% are failing now with a new assertion.  For instance crossgen of System.Net.WebClient.dll
```
Assert failure(PID 21792 [0x00005520], Thread: 21792 [0x5520]): Assertion failed 'ldftnToken == targetMethod->gtFptrVal.gtLdftnResolvedToken' in 'System.Threading.Tasks.BeginEndAwaitableAdapter:.cctor()' (IL size 22)

    File: /home/vmjenkins/workspace/Dotnet/build_and_test/src/jit/flowgraph.cpp Line: 7225
    Image: /home/vmjenkins/workspace/Dotnet/build_and_test/bin/tests/pr1/Tests/coreoverlay/crossgen
```

Here is the exhaustive list of failing libraries.
```
cijobs.dll
jit-analyze.dll
jit-dasm.dll
Microsoft.CodeAnalysis.CSharp.dll
Microsoft.CodeAnalysis.dll
Microsoft.CodeAnalysis.VisualBasic.dll
Microsoft.CSharp.dll
Microsoft.Diagnostics.Tracing.TraceEvent.dll
Microsoft.DotNet.Cli.Utils.dll
Microsoft.DotNet.InternalAbstractions.dll
Microsoft.DotNet.ProjectModel.dll
Microsoft.Extensions.DependencyModel.dll
Microsoft.VisualBasic.dll
Microsoft.Win32.Registry.dll
Newtonsoft.Json.dll
NuGet.Common.dll
NuGet.Configuration.dll
NuGet.DependencyResolver.Core.dll
NuGet.Frameworks.dll
NuGet.LibraryModel.dll
NuGet.Packaging.Core.dll
NuGet.Packaging.Core.Types.dll
NuGet.Packaging.dll
NuGet.ProjectModel.dll
NuGet.Protocol.Core.Types.dll
NuGet.Protocol.Core.v3.dll
NuGet.Repositories.dll
NuGet.RuntimeModel.dll
NuGet.Versioning.dll
System.Collections.dll
System.Collections.Immutable.dll
System.CommandLine.dll
System.ComponentModel.Annotations.dll
System.ComponentModel.EventBasedAsync.dll
System.ComponentModel.TypeConverter.dll
System.Console.dll
System.Data.Common.dll
System.Diagnostics.Debug.dll
System.Diagnostics.DiagnosticSource.dll
System.Diagnostics.Process.dll
System.Diagnostics.StackTrace.dll
System.Diagnostics.Tracing.dll
System.Drawing.Primitives.dll
System.IO.Compression.dll
System.IO.FileSystem.dll
System.IO.FileSystem.DriveInfo.dll
System.IO.FileSystem.Watcher.dll
System.IO.IsolatedStorage.dll
System.IO.Pipes.dll
System.Linq.dll
System.Linq.Expressions.dll
System.Linq.Parallel.dll
System.Linq.Queryable.dll
System.Net.Http.dll
System.Net.HttpListener.dll
System.Net.Mail.dll
System.Net.NameResolution.dll
System.Net.NetworkInformation.dll
System.Net.Ping.dll
System.Net.Primitives.dll
System.Net.Requests.dll
System.Net.Security.dll
System.Net.Sockets.dll
System.Net.WebClient.dll
System.Net.WebSockets.Client.dll
System.ObjectModel.dll
System.Private.DataContractSerialization.dll
System.Private.Uri.dll
System.Private.Xml.dll
System.Private.Xml.Linq.dll
System.Reflection.DispatchProxy.dll
System.Reflection.Metadata.dll
System.Runtime.Extensions.dll
System.Runtime.Serialization.Formatters.dll
System.Security.Claims.dll
System.Security.Cryptography.Algorithms.dll
System.Security.Cryptography.Encoding.dll
System.Security.Cryptography.OpenSsl.dll
System.Security.Cryptography.Primitives.dll
System.Security.Cryptography.X509Certificates.dll
System.Threading.dll
System.Threading.Tasks.Dataflow.dll
System.Threading.Tasks.Parallel.dll
System.Threading.Thread.dll
System.Transactions.Local.dll
xunit.assert.dll
xunit.core.dll
xunit.execution.dotnet.dll
xunit.performance.execution.dll
xunit.performance.metrics.dll
xunit.runner.utility.dotnet.dll
```

**Comment by sdmaclea** [comment]
This appears related to the recent commit #10663 @sandreenko @dotnet/jit-contrib

**Comment by sdmaclea** [comment]
I only tested on [Arm64/Unix] hence the title.  This may affect other targets and platforms since crossgen does not seem be run regularly in CI

**Comment by sandreenko** [comment]
It was fixed yesterday #10738.

**Comment by sdmaclea** [comment]
@sandreenko Thanks.  I confirmed it is fixed on the tip.
